src/io/parquet/read/deserialize/binary/basic.rs

```diff
@@ -480,9 +480,13 @@ impl<'a, O: Offset> utils::Decoder<'a> for BinaryDecoder<O> {
 
 pub(super) fn finish<O: Offset, A: TraitBinaryArray<O>>(
     data_type: &DataType,
-    values: Binary<O>,
-    validity: MutableBitmap,
+    mut values: Binary<O>,
+    mut validity: MutableBitmap,
 ) -> Result<A> {
+    values.offsets.0.shrink_to_fit();
```

Comment:
What's very interesting is:

If I remove these three lines of `shrink_to_fit`, the benchmark performance will drop back into `873ms` using the scripts I comment on below.

Reply:
That is strange. Hoe often is this called? Do we have many small binary buffers?

Reply:
It's only one rowgroup and one page per file.

```
column ["url"]: num_of_pages 1, num_of_rows 106742, size 4896486, size(uncompressed) 11736921
```

Reply:
I did a local benchmark and could find no difference with or without `shrink_to_fit`. I think it might be valid to keep it here as we deserialize in a tight loop and don't do any other heap allocations on this thread (that I am aware of) so it probably is on the end of the memory allocator. 

This should be the case if we realloc at 100 items and the offsets have been properly pre-allocated.

Reply:
That's strange, I think the benchmark needs to be on parallel to see the difference.